tensorflow_datasets/image/nyu_depth_v2.py

```diff
@@ -0,0 +1,115 @@
+"""NYU Depth V2 Dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import csv
+import os
+import numpy as np
+import tensorflow as tf
+
+import tensorflow_datasets as tfds
+
+
+_CITATION = """\
+@inproceedings{Silberman:ECCV12,
+  author    = {Nathan Silberman, Derek Hoiem, Pushmeet Kohli and Rob Fergus},
+  title     = {Indoor Segmentation and Support Inference from RGBD Images},
+  booktitle = {ECCV},
+  year      = {2012}
+}
+@article{Alhashim2018,
+  author    = {Ibraheem Alhashim and Peter Wonka},
+  title     = {High Quality Monocular Depth Estimation via Transfer Learning},
+  journal   = {arXiv e-prints},
+  volume    = {abs/1812.11941},
+  year      = {2018},
+  url       = {https://arxiv.org/abs/1812.11941},
+  eid       = {arXiv:1812.11941},
+  eprint    = {1812.11941}
+}
+"""
+
+
+_DESCRIPTION = """\
+The NYU-Depth V2 data set is comprised of video sequences from a variety of
+indoor scenes as recorded by both the RGB and Depth cameras from the
+Microsoft Kinect.
+"""
+
+
+_URL = 'https://s3-eu-west-1.amazonaws.com/densedepth/nyu_data.zip'
+
+
+def _generate_image_pairs(file_list):
+  with tf.io.gfile.GFile(file_list) as f:
+    return csv.reader(f, delimiter=',')
+
+
+def _load_image(path):
+  with tf.io.gfile.GFile(path, "rb") as fp:
+    image = tfds.core.lazy_imports.PIL_Image.open(fp)
+  return np.array(image)
+
+
+def _normalize_depth(image):
```

Comment:
We usually try to provide the dataset as close as the original one, without any pre-processing. Why is it required ? Does the original dataset mix int8 and int16 images ? If so please add a comment explaining.

It seems weird to me that a dataset would provide a mix of uint8 and uint16 images as it would be really error prone. Is it documented somewhere on their website ? https://cs.nyu.edu/~silberman/datasets/nyu_depth_v2.html